central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -360,13 +408,24 @@ func (m *managerImpl) HandleScanRemove(scanID string) error {
 	return nil
 }
 
-func (m *managerImpl) getWatcher(sensorCtx context.Context, id string) watcher.ScanWatcher {
+func (m *managerImpl) getWatcher(sensorCtx context.Context, id string, numChecks int) watcher.ScanWatcher {
 	var scanWatcher watcher.ScanWatcher
 	concurrency.WithLock(&m.watchingScansLock, func() {
 		var found bool
-		if scanWatcher, found = m.watchingScans[id]; !found {
+		// The check for `numChecks == 0` is here to prevent starting a watcher twice per scan.
+		// It may happen that additional status updates (e.g., state) from CO arrive
+		// after the watcher is removed from the watchingScans (i.e., we have all the checks).
+		// Not checking that would cause a new watcher to be created here and in some circumstances
+		// (when no e-mail is provided for notification), the watcher would time-out and delete the data from DB.
+		if scanWatcher, found = m.watchingScans[id]; !found && numChecks == 0 {
+			// check the scan annotation and do not create watcher if totalChecks is >0 and numCheckResults is 0
 			scanWatcher = watcher.NewScanWatcher(m.automaticReportingCtx, sensorCtx, id, m.readyQueue)
 			m.watchingScans[id] = scanWatcher
+			m.watchingScansStartTime[id] = time.Now()
+			m.updateMaxNumScansRunningInParallelNoLock()
```

Comment:
Since we call to `updateMaxNumScansRunningInParallelNoLock` in the defer call, I think calling it here is not necessary.

Reply:
I'd rather keep it here and remove from the deferred statement - that would save us one locking of the mutex.

Reply:
Removed the deferred in [4ff5709](https://github.com/stackrox/stackrox/pull/15517/commits/4ff5709d89ba20bc91a10aa2300afcce419fce54)